src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/AnalysisDetails.java

```diff
@@ -170,6 +174,39 @@ public String createAnalysisSummary(FormatterFactory formatterFactory) {
         return formatterFactory.documentFormatter().format(document, formatterFactory);
     }
 
+    public String createAnalysisIssueSummary(PostAnalysisIssueVisitor.ComponentIssue componentIssue, FormatterFactory formatterFactory) {
+        final DefaultIssue issue = componentIssue.getIssue();
+
+        String baseImageUrl = configuration.get(CommunityBranchPluginConstants.IMAGE_URL_BASE)
+                .orElse("https://raw.githubusercontent.com/mc1arke/sonarqube-community-branch-plugin/master/src/main/resources/pr-decoration-images");
+
+        Document document = new Document(
+                new Paragraph(new Text(String.format("Type: %s", issue.type().name()))),
+                new Paragraph(new Image(issue.severity(), String.format("%s/checks/Severity/%s.svg?sanitize=true", baseImageUrl, issue.severity().toLowerCase())), new Text(issue.severity())),
+                new Paragraph(new Text(String.format("Message: %s", issue.getMessage())))
+                );
+
+        Long effort = issue.effortInMinutes();
+        if (effort != null) {
+            document.addChild(new Paragraph(new Text(String.format("Duration (min): %s", effort))));
+        }
+        String resolution = issue.resolution();
+        if (StringUtils.isNotBlank(resolution)) {
+            document.addChild(new Paragraph(new Text(String.format("Resolution: %s ", resolution))));
+        }
+        return formatterFactory.documentFormatter().format(document, formatterFactory);
+    }
+
+    public String getSCMPathForIssue(PostAnalysisIssueVisitor.ComponentIssue componentIssue) {
```

Comment:
This should really have a return type of `Optional<String>` and return `Optional.empty()` if the Component Type isn't a `File`. That way any caller can be clear if a path actually existed.